### PR TITLE
Exiting with 1 when using --verify with missing files (fixes #93)

### LINKED ISSUE #93: Return error code when file is renamed or moved
When using the `--verify` flag, if a file being referenced is renamed or moved, `embedme` won't return a non-zero error code. This would be helpful in CI, to ensure that the code being referenced actually exists.

_index.ts_
```ts
function add(x: number, y: number): number {
  return x + y
}
```

_README.md_
`````
```ts
// ./index.ts

function add(x: number, y: number): number {
  return x + y
}
```
`````

```bash
mv index.ts add.ts # rename index.ts to add.ts
npx embedme README.md --verify # run embedme with --verify
echo $? # get return code
```

### Current Behavior

```bash
Verifying...
  Analysing README.md...
   README.md#L1-L7 Found filename ./index.ts in comment in first line, but file does not exist at /embedme-test/index.ts!
0
```

### Desired Behavior

```bash
Verifying...
  Analysing README.md...
   README.md#L1-L7 Found filename ./index.ts in comment in first line, but file does not exist at /embedme-test/index.ts!
1
```

I'm free to open a PR if you're open to it.

### PR DESCRIPTION
**Pull request by jci-aws**
Fixes #93

### CODE REVIEW

File: src/embedme.ts [hunk 1]
```diff
@@ -97,13 +97,17 @@ sourceFiles.forEach((source, i) => {
...
-  const outText = embedme(sourceText, resolvedPath, options);
+  const { outText, error } = embedme(sourceText, resolvedPath, options);
 
   if (options.verify) {
     if (sourceText !== outText) {
       errorLog(chalk => chalk.red(`Diff detected, exiting 1`));
       process.exit(1);
     }
+
+    if (error) {
```
Comment (zakhenry) [inline src/embedme.ts:108]: this should move outside of the `if (options.verify) {` block - if you're doing a regular run, or dry run etc it should still `exit(1)` if there is an error regardless of whether or not the --verify flag was passed.

**Review by zakhenry — CHANGES_REQUESTED** [review]

**Comment by zakhenry** [comment]
Hey @jci-aws thanks for your contribution! One small fixup then it is good to go :)

File: src/embedme.ts (hunk 1 above)
Comment (jci-aws) [reply src/embedme.ts]: Hi @zakhenry, just made this change. I did need to modify the test fixtures slightly, let me know if it looks ok. Thanks!

Comment (zakhenry) [reply src/embedme.ts]: hey thanks for the quick edit, I think the fixture change has reduced some coverage however. I understand that the change was needed as I guess the original checks were failing due to it now exiting with an error code, but there is now no longer a test to cover the error messages emitted on those failing cases.

Would you like me to take a look?

Comment (jci-aws) [reply src/embedme.ts]: Got it. Yeah if you'd like to check it out that'd be great, not sure of the best way to test those...

Comment (zakhenry) [reply src/embedme.ts]: ok cool I've fixed that up in #95 , this replaces this PR as I took your commits and added one of my own.

Note this won't auto deploy to npm just yet as I will need to migrate from travis ci to github actions which I will complete separately.

**Pull request closed without merging**